examples/window-app/silabs/efr32/BUILD.gn

```diff
@@ -19,57 +19,85 @@ import("//build_overrides/pigweed.gni")
-examples_plat_dir = "${chip_root}/examples/platform/silabs/efr32"
+silabs_project_dir = "${project_dir}/silabs/efr32"
 examples_common_plat_dir = "${chip_root}/examples/platform/silabs"
-if (chip_enable_ble_rs911x) {
-  src_plat_dir = "${chip_root}/src/platform/silabs/efr32"
+
+if (wifi_soc) {
+  import("${chip_root}/third_party/silabs/SiWx917_sdk.gni")
+  examples_plat_dir = "${chip_root}/examples/platform/silabs/SiWx917"
+} else {
+  import("${efr32_sdk_build_root}/efr32_sdk.gni")
+  examples_plat_dir = "${chip_root}/examples/platform/silabs/efr32"
 }
 
-import("${examples_plat_dir}/args.gni")
+import("${examples_common_plat_dir}/args.gni")
 
 declare_args() {
   # Dump memory usage at link time.
   chip_print_memory_usage = false
 }
 
-efr32_sdk("sdk") {
-  sources = [
-    "${efr32_project_dir}/include/CHIPProjectConfig.h",
-    "${examples_plat_dir}/FreeRTOSConfig.h",
-  ]
+if (wifi_soc) {
+  siwx917_sdk("sdk") {
+    sources = [
+      "${examples_plat_dir}/FreeRTOSConfig.h",
+      "${silabs_project_dir}/include/CHIPProjectConfig.h",
+    ]
 
-  include_dirs = [
-    "${chip_root}/src/platform/silabs/efr32",
-    "${efr32_project_dir}/include",
-    "${examples_plat_dir}",
-    "${chip_root}/src/lib",
-    "${examples_common_plat_dir}",
-  ]
+    include_dirs = [
+      "${chip_root}/src/platform/silabs/SiWx917",
+      "${silabs_project_dir}/include",
+      "${examples_plat_dir}",
+      "${chip_root}/src/lib",
+      "${examples_common_plat_dir}",
+    ]
 
-  if (use_wf200) {
-    # TODO efr32_sdk should not need a header from this location
-    include_dirs += [ "${examples_plat_dir}/wf200" ]
+    defines = []
+    if (chip_enable_pw_rpc) {
+      defines += [
+        "HAL_VCOM_ENABLE=1",
+        "PW_RPC_ENABLED",
+      ]
+    }
   }
+} else {
+  efr32_sdk("sdk") {
+    sources = [
+      "${examples_plat_dir}/FreeRTOSConfig.h",
+      "${silabs_project_dir}/include/CHIPProjectConfig.h",
+    ]
 
-  if (chip_enable_ble_rs911x) {
-    # TODO efr32_sdk should not need a header from this location
-    include_dirs += [
-      "${src_plat_dir}/rs911x",
-      "${examples_plat_dir}/rs911x",
-      "${examples_plat_dir}/rs911x/hal",
+    include_dirs = [
+      "${chip_root}/src/platform/silabs/efr32",
+      "${silabs_project_dir}/include",
+      "${examples_plat_dir}",
+      "${chip_root}/src/lib",
+      "${examples_common_plat_dir}",
     ]
+
+    if (use_wf200) {
+      # TODO efr32_sdk should not need a header from this location
+      include_dirs += [ "${examples_plat_dir}/wf200" ]
+    }
+
+    if (chip_enable_ble_rs911x) {
+      # TODO efr32_sdk should not need a header from this location
+      include_dirs += [
+        "${chip_root}/src/platform/silabs/efr32/rs911x",
+        "${examples_plat_dir}/rs911x",
+        "${examples_plat_dir}/rs911x/hal",
+      ]
+    }
   }
 }
 
 silabs_executable("window_app") {
-  output_name = "chip-efr32-window-example.out"
+  output_name = "chip-silabs-window-example.out"
```

Comment:
do we want to keep `chip_` might as well change it while we are here.

Reply:
changed